### DISCUSSION TITLE
Does yarp use the incoming port on destinations?

### DISCUSSION THREAD

**Discussion by Marbry** [dotnet/yarp#2388] [General] [answered]
If I'm using a Hosts filter on a route like `"Hosts": "mydnsname:5000"`, and I don't specify a port on the cluster destinations such as 
```
"Address": "https://webnode1.mydomain"
"Address": "https://webnode2.mydomain"
``` 
does yarp use the incoming port of 5000 on the destination selected in the cluster?  Or does it have to be explicitly mapped for every single port, where each port is a separate application?

**Comment by MihaZupan** [comment, marked as answer]
No, YARP will use the address prefix as-is, the port is not copied over.

You can either specify the exact ports you want to use on the backend, or you can use a transform to modify requests, e.g.
```c#
builder.Services.AddReverseProxy()
    .LoadFromMemory(GetRoutes(), GetClusters())
    .AddTransforms(transforms =>
    {
        transforms.AddRequestTransform(context =>
        {
            var uri = RequestUtilities.MakeDestinationAddress(context.DestinationPrefix, context.Path, context.Query.QueryString);
            uri = new UriBuilder(uri) { Port = context.HttpContext.Connection.LocalPort }.Uri;
            context.ProxyRequest.RequestUri = uri;
            return default;
        });
    });
```

will set the destination port to the connection's local port.

**Comment by Marbry** [comment]
Thanks, I didn't want to have to define a separate cluster for every single app, since they're all the same servers, but it looks like I need to define the destinations with the port for each one then.  I wasn't doing it in code because I may be adding things that don't follow the same pattern.

**Reply by Tratcher** [reply]
Code can be conditional. You can include metadata on routes, clusters, and destinations for dynamic consumption at runtime.

**Reply by Marbry** [reply]
Sure, I understand, but in this case it's easier to just have it in the settings I think.  Thanks!
